x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/label_node/label_node.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { Handle, Position } from '@xyflow/react';
+import { css } from '@emotion/react';
+import { EuiText, EuiTextTruncate, EuiToolTip, useEuiShadow, useEuiTheme } from '@elastic/eui';
+import {
+  LabelNodeContainer,
+  LabelShape,
+  HandleStyleOverride,
+  LabelShapeOnHover,
+  NodeButton,
+  ACTUAL_LABEL_HEIGHT,
+  NODE_LABEL_WIDTH,
+  getLabelColors,
+} from '../styles';
+import type { LabelNodeViewModel, NodeProps } from '../../types';
+import { NodeExpandButton } from '../node_expand_button';
+import { analyzeDocuments } from './analyze_documents';
+import { LabelNodeBadges, LIMIT as BADGES_LIMIT } from './label_node_badges';
+import { LabelNodeTooltipContent } from './label_node_tooltip';
+import { LabelNodeDetails } from './label_node_details';
+
+export const TEST_SUBJ_CONTAINER = 'label-node-container';
+export const TEST_SUBJ_SHAPE = 'label-node-shape';
+export const TEST_SUBJ_TOOLTIP = 'label-node-tooltip';
+export const TEST_SUBJ_HANDLE = 'label-node-handle';
+export const TEST_SUBJ_EXPAND_BTN = 'label-node-expand-btn';
+export const TEST_SUBJ_HOVER_OUTLINE = 'label-node-hover-outline';
+
+export const LabelNode = memo<NodeProps>((props: NodeProps) => {
+  const {
+    id,
+    color,
+    eventsCount,
+    alertsCount,
+    label,
+    interactive,
+    ips,
+    countryCodes,
+    nodeClick,
+    expandButtonClick,
+  } = props.data as LabelNodeViewModel;
+
+  const { euiTheme } = useEuiTheme();
+  const shadow = useEuiShadow('m', { property: 'filter' });
+
+  const text = label ? label : id;
+
+  const { backgroundColor, borderColor, textColor } = useMemo(
+    () => getLabelColors(color, euiTheme),
+    [color, euiTheme]
+  );
+
+  const numEvents = eventsCount ?? 0;
+  const numAlerts = alertsCount ?? 0;
+  // TODO Remove color when API returns `eventsCount` and `alertsCount`
+  const analysis = analyzeDocuments({ color, eventsCount: numEvents, alertsCount: numAlerts });
+  const shouldShowTooltip = numEvents > BADGES_LIMIT || numAlerts > BADGES_LIMIT;
```

Comment:
we don't show tooltip when the text is trimmed? if so, a friendly reminder to have something in the backlog to remind us to get back to it